### PR TITLE
Admin NFT STM: votes with single value instead of 2 separate yes no lists

### PR DESCRIPTION
**Pull request by RKlompUU**
votes as a single underlying yes or no value per user. That is, 1 user can only have a yes vote, or a no vote, or no vote at all for a specific attribute. (before it was possible to vote yes, and to vote no on something at the same time.

not to be reviewed until this PR has been merged though, as it's branched from there: https://github.com/tzConnectBerlin/kanvas/pull/66

### CODE REVIEW

File: admin-api-server/src/nft/service/nft.service.ts [hunk 1]
```diff
@@ -250,22 +251,31 @@ RETURNING id
...
-      );
+
+    try {
+      const nft = await this.#findOne(nftId);
+      if (nft.createdBy !== user.id) {
+        throw new HttpException(
+          'no permission to delete this nft (only the creator may)',
+          HttpStatus.FORBIDDEN,
+        );
+      }
```
Comment (TristanNcl) [inline admin-api-server/src/nft/service/nft.service.ts:262]: I know we said only creator can delete the nft in the creation phase, but looking at this now make me wonder if adding the admin make sense as well

File: admin-api-server/lib/src/stm.ts [hunk 2]
```diff
@@ -124,59 +118,103 @@ export class StateTransitionMachine {
...
+    return unmetTransitionConditions;
   }
 }
+
+function isNone(v: any): boolean {
+  // note: v here is checked for both null and undefined
+  //       because undefined coerces into null. And its safe because nothing
+  //       else coerces into null (other than null itself).
+  return v == null;
+}
```
Comment (TristanNcl) [inline admin-api-server/lib/src/stm.ts:220]: Bravo !

**Review by TristanNcl — APPROVED** [review]

File: admin-api-server/src/nft/service/nft.service.ts (hunk 1 above)
Comment (RKlompUU) [reply admin-api-server/src/nft/service/nft.service.ts:262]: Yeah I think it does too. I'll add it.

File: admin-api-server/lib/src/stm.ts (hunk 2 above)
Comment (RKlompUU) [reply admin-api-server/lib/src/stm.ts:220]: If the codeline itself makes little sense while reading it, can always put it into a helper function that is named properly haha

File: admin-api-server/config/e2e_tests.yaml [hunk 3]
```diff
@@ -9,41 +9,48 @@ attributes:
...
-      when: |
-        nft.publish &&
-        nft['image.png'].length > 0 &&
-        nft['thumbnail.png'].length > 0 &&
-        nft.description.length > 0
+      when:
+      - nft.publish['yes'].length >= 2
+      - nft['image.png'].length > 0
+      - nft['thumbnail.png'].length > 0
+      - nft.description.length > 0
```
[the remark this answers is not in the document]
Comment (RKlompUU) [reply admin-api-server/config/e2e_tests.yaml:46]: This is just the config file for the end to end tests! So isn't related to deployment

**Pull request merged**